src/Symfony/Component/Serializer/Normalizer/ProblemNormalizer.php

```diff
@@ -53,15 +63,47 @@ public function normalize(mixed $object, string $format = null, array $context =
             throw new InvalidArgumentException(sprintf('The object must implement "%s".', FlattenException::class));
         }
 
-        $context += $this->defaultContext;
+        $data = [];
         $debug = $this->debug && ($context['debug'] ?? true);
+        $exception = $context['exception'] ?? null;
+        if ($exception instanceof HttpExceptionInterface) {
+            $exception = $exception->getPrevious();
+
+            if ($exception instanceof PartialDenormalizationException) {
+                $trans = $this->translator ? $this->translator->trans(...) : fn ($m, $p) => strtr($m, $p);
+                $template = 'This value should be of type {{ type }}.';
+                $data = [
+                    self::TYPE => 'https://symfony.com/errors/validation',
+                    'violations' => array_map(
+                        fn ($e) => [
+                            'propertyPath' => $e->getPath(),
+                            'title' => $trans($template, [
+                                '{{ type }}' => implode('|', $e->getExpectedTypes() ?? ['?']),
+                            ], 'validators'),
+                            'template' => $template,
+                            'parameter' => [
+                                '{{ type }}' => implode('|', $e->getExpectedTypes() ?? ['?']),
+                            ],
+                        ] + ($debug || $e->canUseMessageForUser() ? ['hint' => $e->getMessage()] : []),
+                        $exception->getErrors()
+                    ),
+                ];
+            } elseif ($exception instanceof ValidationFailedException
+                && $this->serializer instanceof NormalizerInterface
+                && $this->serializer->supportsNormalization($exception->getViolations(), $format, $context)
+            ) {
+                $data = $this->serializer->normalize($exception->getViolations(), $format, $context);
+            }
+        }
+
+        $context += $this->defaultContext;
```

Comment:
moving this after the usage of `$context['debug'] ?? true` looks wrong to me.

Reply:
should be good now